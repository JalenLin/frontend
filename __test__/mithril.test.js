import m from "mithril";
import { tidy } from "mithril-tidy";
import Component from "../src/app/components/misc/Searchbar";

test("can render the searchbar", () => {
    const selector = {
        get: () => "new mock data",
        set: jest.fn()
    };
    const comp = m(Component, { selector: selector });
    const html = tidy(comp);
    expect(html).toMatchSnapshot();
});