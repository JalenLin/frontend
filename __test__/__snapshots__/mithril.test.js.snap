exports[`test can render the searchbar 1`] = `
"<div class=\"search-form\">
  <div class=\"search-input\">
    <input type=\"text\"><span><input type=\"checkbox\" id=
    \"exactSearch\"><label for=\"exactSearch\">exact
    name</label></span>
  </div><button class=\"search-submit\">Search</button>
</div>
"
`;
